kitsune/wiki/views.py

```diff
@@ -71,6 +71,15 @@ def _doc_page_cache_view(request, document_slug, *args, **kwargs):
             statsd.incr('wiki.document_view.cache.skip')
             return view(request, document_slug, *args, **kwargs)
 
+        # If the slug is part of the UX experiments, redirect to the experiment view.
+        # The check is here because the KB articles are heavily cached and we need to load
+        # balance based on the waffle flag for every request.
+        if (switch_is_active('ux_experiment_1') and
+            flag_is_active(request, 'ux_experiment_1') and
+            all([x not in request.META['HTTP_USER_AGENT'] for x in EXCLUDED_BROWSERS]) and
+                document_slug in SUMO_UX_EXPERIMENTS_SLUGS and request.LANGUAGE_CODE == 'en-US'):
```

Comment:
It would be better to move the slug and language check first so that the DB is not checked for the switch status for every document page load.